packages/web3-eth-abi/src/utils.ts

```diff
@@ -123,12 +123,26 @@ export const mapTypes = (
 	return mappedTypes;
 };
 
+/**
+ * returns true if input is a hexstring and is odd-lengthed
+ */
+export const isOddHexstring = (param: unknown): boolean =>
+	typeof param === 'string' && /^(-)?0x[0-9a-f]*$/i.test(param) && param.length % 2 === 1;
+
+/**
+ * format odd-length bytes to even-length
+ */
+export const formatOddHexstrings = (param: string): string =>
+	isOddHexstring(param) ? `0x0${param.substring(2)}` : param;
+
 /**
  * Handle some formatting of params for backwards compatibility with Ethers V4
  */
 export const formatParam = (type: string, _param: unknown): unknown => {
 	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-	let param = _param;
+
+	// clone if _param is an object
+	const param = typeof _param === 'object' && !Array.isArray(_param) ? { ..._param } : _param;
```

Comment:
@luu-alex  This PR LGTM, one observation: Even its array we should clone it I think. or is there any reason for not doing that?

Reply:
nevermind, i updated to cloning if its an array

Reply:
https://github.com/ChainSafe/web3.js/pull/4882/files#diff-2147b746d8cf05a0c42e16fa6d2aa4fa081357e50af4ca6d22c1487ee3e5a44fR158